pom.xml

```diff
@@ -4,32 +4,206 @@
+            <configuration> <includes> <include>**/Test*.java</include> <include>**/*Test.java</include> </includes> <excludes> <exclude>**/*FuncTest.java</exclude> </excludes> <forkCount>1</forkCount> <reuseForks>false</reuseForks> <systemPropertyVariables>
+            <environment>${environment}</environment> <site>${site}</site> </systemPropertyVariables> </configuration> </plugin> -->
+
+            <!-- log4j audit maven plugin -->
+            <plugin>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-audit-maven-plugin</artifactId>
+                <version>${log4j-audit.version}</version>
+                <executions>
+                    <execution>
+                        <id>generate</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>generate</goal>
+                        </goals>
+                        <configuration>
+                            <packageName>org.apache.logging.log4j.audit.event</packageName>
+                            <enterpriseId>18060</enterpriseId>
+                            <maxKeyLength>32</maxKeyLength>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <!-- versions-maven-plugin -->
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>versions-maven-plugin</artifactId>
+                <version>${versions-maven-plugin.version}</version>
+                <configuration>
+                    <generateBackupPoms>false</generateBackupPoms>
+                </configuration>
+            </plugin>
+            <!-- maven enforcer and complier -->
+            <!-- <plugin> <groupId>org.apache.maven.plugins</groupId> <artifactId>maven-compiler-plugin</artifactId> <version>3.8.0</version> <configuration> <source>[1.8,)</source> <target>[1.8,)</target> <showDeprecation>true</showDeprecation>
+            <showWarnings>true</showWarnings> <compilerArguments> <Xmaxwarns>10000</Xmaxwarns> <Xlint/> </compilerArguments> <encoding>UTF-8</encoding> </configuration> </plugin> -->
+            <plugin>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <version>${maven-enforcer-plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>display-info</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>display-info</goal>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireMavenVersion>
```

Comment:
You don't need to (and shouldn't) be redefining all the rules. Just override the broken one as documented in [the Jenkins plugin parent pom](https://github.com/jenkinsci/plugin-pom/blob/master/pom.xml#L536-L552).

Reply:
I've just refactored the pom.xml

An update of the `extra-enforcer-rules` version fixes the banned dependency issue. Also eliminates the need to have to [add the extra lines](https://github.com/jenkinsci/plugin-pom/blob/master/pom.xml#L536-L552_

Reply:
You still have a ton of unnecessary lines in this rules block.